src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,132 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [formData, setFormData] = useState({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
+
+  const [, setErrors] = useState({
+    title: false,
+    imgUrl: false,
+    imdbUrl: false,
+    imdbId: false,
+  });
+
+  type FormDataKeys = 'title' | 'description' | 'imgUrl' | 'imdbUrl' | 'imdbId';
+
+  const handleBlur = (field: FormDataKeys) => {
+    setErrors(prevErrors => ({
+      ...prevErrors,
+      [field]: !formData[field].trim(),
+    }));
+  };
+
+  const handleChange = (field: FormDataKeys, value: string) => {
+    setFormData(prevData => ({
+      ...prevData,
+      [field]: value,
+    }));
+  };
+
+  const resetFormAfterSubmit = () => {
+    setFormData({
+      title: '',
+      description: '',
+      imgUrl: '',
+      imdbUrl: '',
+      imdbId: '',
+    });
+    setCount(prevCount => prevCount + 1);
+  };
+
+  const isFormValid = () => {
+    const { title, imdbId, imdbUrl, imgUrl } = formData;
+
+    return title.trim() && imdbId.trim() && imdbUrl.trim() && imgUrl.trim();
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!isFormValid) {
```

Comment:
You are missing parentheses to invoke the function.
```suggestion
    if (!isFormValid()) {
```